native/membrane_videocompositor/src/compositor/texture_transformations/corners_rounding/corners_rounding.wgsl

```diff
@@ -0,0 +1,98 @@
+struct VertexInput {
+    @location(0) position: vec3<f32>,
+    @location(1) texture_coords: vec2<f32>
+}
+
+struct VertexOutput {
+    @builtin(position) position: vec4<f32>,
+    @location(0) texture_coords: vec2<f32>,
+}
+
+@vertex
+fn vs_main(input: VertexInput) -> VertexOutput {
+    var output: VertexOutput;
+
+    output.position = vec4<f32>(input.position, 1.0);
+    output.texture_coords = input.texture_coords;
+
+    return output;
+}
+
+struct CornersRoundingUnifrom{
+    border_radius: f32,
+    // video_width: f32,
+    // video_height: f32
+}
+
+@group(0) @binding(0)
+var texture: texture_2d<f32>;
+
+@group(1) @binding(0)
+var sampler_: sampler;
+
+@group(2) @binding(0)
+var<uniform> corners_rounding_uniform: CornersRoundingUnifrom;
+
+struct IsInCorner {
+    left_border: bool,
+    right_border: bool,
+    top_border: bool,
+    bot_border: bool,
+}
+
+fn get_nearest_inner_corner_coords_in_pixels(
+    is_on_edge: IsInCorner,
+    video_width: f32,
+    video_height: f32,
+    border_radius: f32
+) -> vec2<f32> {
+    if (is_on_edge.left_border && is_on_edge.top_border) {
+        return vec2<f32>(border_radius, border_radius);
+    } else if (is_on_edge.right_border && is_on_edge.top_border) {
+        return vec2<f32>(video_width - border_radius, border_radius);
+    } else if (is_on_edge.right_border && is_on_edge.bot_border) {
+        return vec2<f32>(video_width - border_radius, video_height - border_radius);
+    } else {
+        return vec2<f32>(border_radius, video_height - border_radius);
+    }
+}
+
+@fragment
+fn fs_main(input: VertexOutput) -> @location(0) vec4<f32> {
+    // Firstly calculates, whether the pixel is in the square in one of the video corners,
+    // then calculates the distance to the center of the circle located in corner of the video
+    // and if the distance is larger than the circle radius, it makes the pixel transparent.
+
+    let border_radius = corners_rounding_uniform.border_radius;
+    let video_resolution = vec2<f32>(textureDimensions(texture));
+    let video_width = video_resolution.x;
+    let video_height = video_resolution.y;
```

Comment:
isn't this actually wrong? this is the _input_ resolution, and I thought we agreed that rounding should be done according to the output dimensions?

Reply:
Ah right, I forgot, that if this transformation is the first one, then this texture hasn't been rendered yet on plane with base size. Thinking of that, do you think you should keep passing `video_width` and `video_height` to shader buffer? In order for that to work, we would need to firstly render frame without any transformations to frame of output placement (with base resolution). I think passing those parameters to shader buffer is better solution. What do you think?

Reply:
Yeah, I sadly think we need to send the width and height to the buffer

Reply:
Ok, done